File: app/schemas/history.py
from datetime import datetime

from pydantic import BaseModel, Field


class AnalysisHistoryEntry(BaseModel):
    id: int = Field(..., ge=1)
    analyzed_at: datetime
    youtube_video_id: str
    video_url: str
    video_title: str | None
    positive_count: int
    neutral_count: int
    negative_count: int
    total_predictions: int


class AnalysisHistoryListResponse(BaseModel):
    items: list[AnalysisHistoryEntry]
    total: int

File: web/src/types/history.ts
export interface AnalysisHistoryEntry {
  id: number;
  analyzed_at: string;
  youtube_video_id: string;
  video_url: string;
  video_title: string | null;
  positive_count: number;
  neutral_count: number;
  negative_count: number;
  total_predictions: number;
}

export interface AnalysisHistoryListResponse {
  items: AnalysisHistoryEntry[];
  total: number;
}
